pythran/analyses/aliases.py

```diff
@@ -106,55 +95,164 @@ def rec(w, n):
+        (a if c else b) => ['a', 'b']
+        '''
         self.visit(node.test)
         rec = map(self.visit, [node.body, node.orelse])
         return self.add(node, set.union(*rec))
 
     def visit_Dict(self, node):
-        self.generic_visit(node)
-        return self.add(node)  # not very accurate
+        '''
+        A dict is abstracted as an unordered container of its values
+
+        >>> from pythran import passmanager
+        >>> pm = passmanager.PassManager('demo')
+        >>> module = ast.parse('def foo(a, b): return {0: a, 1: b}')
+        >>> result = pm.gather(Aliases, module)
+        >>> Aliases.dump(result, filter=ast.Dict)
+        {0: a, 1: b} => ['|a|', '|b|']
+
+        where the |id| notation means something that may contain ``id``.
+        '''
+        if node.keys:
+            elts_aliases = set()
+            for key, val in zip(node.keys, node.values):
+                self.visit(key)  # res ignored, just to fill self.aliases
+                elt_aliases = self.visit(val)
+                elts_aliases.update(map(ContainerOf, elt_aliases))
+        else:
+            elts_aliases = None
+        return self.add(node, elts_aliases)
 
     def visit_Set(self, node):
-        self.generic_visit(node)
-        return self.add(node)  # not very accurate
+        '''
+        A set is abstracted as an unordered container of its elements
+
+        >>> from pythran import passmanager
+        >>> pm = passmanager.PassManager('demo')
+        >>> module = ast.parse('def foo(a, b): return {a, b}')
+        >>> result = pm.gather(Aliases, module)
+        >>> Aliases.dump(result, filter=ast.Set)
+        {a, b} => ['|a|', '|b|']
+
+        where the |id| notation means something that may contain ``id``.
+        '''
+        if node.elts:
+            elts_aliases = set()
+            for elt in node.elts:
+                elt_aliases = self.visit(elt)
+                elts_aliases.update(map(ContainerOf, elt_aliases))
+        else:
+            elts_aliases = None
+        return self.add(node, elts_aliases)
 
     def visit_Return(self, node):
+        '''
+        A side effect of computing aliases on a Return is that it updates the
+        ``return_alias`` field of current function
+
+        >>> from pythran import passmanager
+        >>> pm = passmanager.PassManager('demo')
+        >>> module = ast.parse('def foo(a, b): return a')
+        >>> result = pm.gather(Aliases, module)
+        >>> module.body[0].return_alias # doctest: +ELLIPSIS
+        <function merge_return_aliases at...>
+
+        This field is a function that takes as many nodes as the function
+        argument count as input and returns an expression based on
+        these arguments if the function happens to create aliasing
+        between its input and output. In our case:
+
+        >>> f = module.body[0].return_alias
+        >>> Aliases.dump(f([ast.Name('A', ast.Load()), ast.Num(1)]))
+        ['A']
```

Comment:
Why would you want one? You know for sure you'll alias  to ``a``, if I'm not mistaken.